### DISCUSSION TITLE
Slow setup on Ubuntu 24.04 PHP version 8.3

### DISCUSSION THREAD

**Discussion by DanielGSoftware** [shivammathur/setup-php#886] [Q&A / Help]
Hello @shivammathur!

We’re moving to runners based on Ubuntu 24.04. While testing this, we noticed the setup-php step takes much longer—about 2-3 minutes compared to the usual 20 seconds on our Ubuntu 22.04 runners.

The biggest delay seems to happen in the first step, setup-php. On Ubuntu 22.04, the message says: PHP Installed PHP 8.3.14, and it’s fast. But on Ubuntu 24.04, it often says: PHP Updated PHP 8.3.14, and it’s slow.

I read in this [discussion](https://github.com/shivammathur/setup-php/issues/63) that delays might happen if the required PHP version isn’t pre-installed. However, Ubuntu 24.04 should already have PHP 8.3. Also, I’m confused because PHP 8.3 works fast on Ubuntu 22.04, even though this [table](https://github.com/shivammathur/setup-php/issues/63) says it’s not pre-installed there either.

Another thing: I've also encountered that on Ubuntu 24.04, the step does finish quickly (about 6 seconds), although this happens almost never. I don’t know why it’s inconsistent.

We’re not in a rush to switch to Ubuntu 24.04, but I’d love to understand what’s going on.

Thank you for this amazing package and for your time!
Looking forward to hearing from you.

**Comment by shivammathur** [comment]
We have builds cached for GitHub hosted runner as we are familiar with the environment, so that is fast as it just involves downloading the build and extracting it. 

On a self-hosted runner it installs the PHP version using apt packages. Also there are dependencies that it checks for on self-hosted runners. So it takes more time.

**Reply by DanielGSoftware** [reply]
Thank you for the quick reply. We aren't using any self-hosted runners, but runners provisioned by GitHub. I see now that my written message "switching to our own runners" is **very** misleading, hence why I think the misinterpretation happened. Sorry for that.

**Reply by shivammathur** [reply]
Ok. On GitHub runners when it says updated, GitHub has a old version installed that comes default with ubuntu, the action updates that to the latest patch version.

**Comment by Rockylars** [comment]
I'm also experiencing this issue. On some runners it will finish PHP 8.2 in 22 seconds and on some it'll be quadruple that. I'm going to downgrade to 22.04 to see if that solves the issue. It happens on multiple repositories and occurs very randomly. Started around a week or two ago, it seems the downgrade has helped.
